src/Symfony/Component/Config/Tests/Definition/ScalarNodeTest.php

```diff
@@ -57,4 +57,51 @@ public function getInvalidValues()
             array(new \stdClass()),
         );
     }
+
+    /**
+     * @dataProvider getValidNonEmptyValues
+     *
+     * @param mixed $value
+     */
+    public function testValidNonEmptyValues($value)
+    {
+        $node = new ScalarNode('test');
+        $node->setAllowEmptyValue(false);
+
+        $this->assertSame($value, $node->finalize($value));
+    }
+
+    public function getValidNonEmptyValues()
+    {
+        return array(
```

Comment:
It's true that we could have added it, but I think it's not absolutely necessary. Of course, if you think it deserves it own test, please don't hesitate to make a pull request for it. :)